src/components/axes/abstractAxis.ts

```diff
@@ -488,16 +488,43 @@ export module Axis {
                                     });
       var lastLabelClientRect: ClientRect;
 
-      visibleTickLabels.each(function (d: any) {
+      // Base interval value between labels
+      var interval = 1;
+
+      // Get the widths of all the labels
+      var visibleTickLabelRects: ClientRect[] = [];
+      visibleTickLabels.each(function(d: any) {
         var clientRect = this.getBoundingClientRect();
+        visibleTickLabelRects.push(clientRect);
+      });
+
+      // Figure out the appropriate interval for overlap
+      var hasSkipped = false;
+      for (var i = 1; i < visibleTickLabelRects.length; i++) {
+        // Checks to see if there is any overlap between the first element and the current element
+        if (!_Util.DOM.boxesOverlap(visibleTickLabelRects[i], visibleTickLabelRects[0])) {
+          // We increment the variable if the next label's width is greater than the previous one's (one-time)
+          // This only applies to the bottom and top orientations
+          if (visibleTickLabelRects[i].width > visibleTickLabelRects[i-1].width && !hasSkipped &&
+            (this._orientation === "bottom" || this._orientation === "top")) {
+            interval += 1;
+            hasSkipped = true;
```

Comment:
So this was an interesting case that I encountered with a fairly narrow axis that went from 0 to 1 in increments of 0.1. There is no overlap between the 0 and the 0.1 label because the 0 label is smaller, but the 0.1 label and the 0.2 label have an overlap due to their size. The skipping covers this edge case and maintains a consistent interval throughout the rendered axis. 

develop:
![image](https://cloud.githubusercontent.com/assets/1175041/5950412/b977641a-a70e-11e4-8fcc-8b639d0065a3.png)

#1556:
![image](https://cloud.githubusercontent.com/assets/1175041/5950399/7c498456-a70e-11e4-95e6-8011c5bf510b.png)